### DISCUSSION TITLE
Vector enabled using Systemctl is throwing Request timeout error

### DISCUSSION THREAD

**Discussion by jeyakumar8** [vectordotdev/vector#18409] [Q&A]
Hi,

I'm running a vector agent in Debian os. Vector pipeline is integrated with prometheus_remote_write sink.

The issue is, whenver i try to enable the vector using /usr/bin/vector. It is able to send data to the prometheus Backend. But if the vector is enabled using systemctl it is not able to send metrics and i'm getting request timed out error.

`vector::sinks::util::retries: Request timed out. If this happens often while the events are actually reaching their destination, try decreasing `batch.max_bytes` and/or using `compression` if applicable. Alternatively `request.timeout_secs` can be increased. internal_log_rate_limit=true`

Anyone have any idea on this behaviour?

PS: I'm using the same vector.toml for both the cases.

Thanks in Advance for your help!!

**Comment by jszwedko** [comment]
Hi @jeyakumar8 !

It sounds like there might be an environment difference when you are running under systemd. Have you verified that, for example, any environment variables are appropriately set when running under systemd? By default the unit file will load them from `/etc/defaults/vector`.
